cirq/linalg/diagonalize.py

```diff
@@ -40,8 +43,9 @@ def diagonalize_real_symmetric_matrix(
         ValueError: Matrix isn't real symmetric.
     """
 
-    # TODO: Determine if thresholds should be passed into is_hermitian
-    if np.any(np.imag(matrix) != 0) or not predicates.is_hermitian(matrix):
+    if check_preconditions and (
+            np.any(np.imag(matrix) != 0) or
```

Comment:
In this case it is important the matrix be _exactly_ real, since otherwise the solution may use a unitary matrix instead of an orthogonal matrix. I suppose I agree that you could assert that all the complex parameters must be less than some threshold and then just trim them away...